usaspending_api/search/v2/views/search.py

```diff
@@ -622,84 +623,67 @@ def post(self, request):
-                except Exception:
-                    raise InvalidParameterException("Invalid field value: {}".format(field))
-
         # build sql query filters
-        queryset = award_filter(filters, UniversalAwardView).values(*values)
+        queryset = award_filter(filters, UniversalAwardView).values()
 
-        # build response
-        response = {"limit": limit, "results": []}
-        results = []
+        values = {'award_id', 'piid', 'fain', 'uri', 'type'}  # always get at least these columns
+        for field in fields:
+            if award_contracts_mapping.get(field):
+                values.add(award_contracts_mapping.get(field))
+            if loan_award_mapping.get(field):
+                values.add(loan_award_mapping.get(field))
+            if non_loan_assistance_award_mapping.get(field):
+                values.add(non_loan_assistance_award_mapping.get(field))
 
         # Modify queryset to be ordered if we specify "sort" in the request
         if sort:
-            sort_filters = []
             if set(filters["award_type_codes"]) <= set(contract_type_mapping):
                 sort_filters = [award_contracts_mapping[sort]]
             elif set(filters["award_type_codes"]) <= set(loan_type_mapping):  # loans
                 sort_filters = [loan_award_mapping[sort]]
             else:  # assistance data
                 sort_filters = [non_loan_assistance_award_mapping[sort]]
+
             if sort == "Award ID":
                 sort_filters = ["piid", "fain", "uri"]
             if order == 'desc':
                 sort_filters = ['-' + sort_filter for sort_filter in sort_filters]
-            if sort_filters:
-                queryset = queryset.order_by(*sort_filters)
+
+            queryset = queryset.order_by(*sort_filters).values(*list(values))
 
         limited_queryset = queryset[lower_limit:upper_limit + 1]
         has_next = len(limited_queryset) > limit
 
+        results = []
         for award in limited_queryset[:limit]:
             row = {"internal_id": award["award_id"]}
-            if set(filters["award_type_codes"]) <= set(contract_type_mapping):
+
+            if award['type'] in contract_type_mapping.keys():
                 for field in fields:
-                    row[field] = award[award_contracts_mapping[field]]
-            elif set(filters["award_type_codes"]) <= set(loan_type_mapping):  # loans
+                    row[field] = award.get(award_contracts_mapping.get(field))
+            elif award['type'] in loan_type_mapping.keys():  # loans
                 for field in fields:
-                    row[field] = award[loan_award_mapping[field]]
-            elif set(filters["award_type_codes"]) <= set(non_loan_assistance_type_mapping):  # assistance data
+                    row[field] = award.get(loan_award_mapping.get(field))
+            elif award['type'] in non_loan_assistance_type_mapping.keys():  # assistance data
```

Comment:
Like the checks here, only comment is that I think in Python you don't need the `.keys()` calls, just checking to see if it's in the dictionary should check if the value is in the dictionary's keys. That being said, the `.keys()` functions are more readable so it really just comes down to a matter of taste.

Reply:
You are correct, the `.keys()` is unnecessary. The best practice also dictates to use it without the `.keys()` so I will update that.